src/Mvc/Mvc.Formatters.Json/src/JsonInputFormatter.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Formatters.Json;
+
+namespace Microsoft.AspNetCore.Mvc.Formatters
+{
+    /// <summary>
+    /// A <see cref="TextInputFormatter"/> for JSON content.
+    /// </summary>
+    public sealed class JsonInputFormatter : TextInputFormatter, IInputFormatterExceptionPolicy
+    {
+        private readonly JsonSerializerOptions _serializerOptions;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="JsonInputFormatter"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="JsonFormatterOptions"/>.</param>
+        public JsonInputFormatter(JsonFormatterOptions options)
+        {
+            _serializerOptions = options.SerializerOptions;
+
+            SupportedEncodings.Add(UTF8EncodingWithoutBOM);
+            SupportedEncodings.Add(UTF16EncodingLittleEndian);
+
+            SupportedMediaTypes.Add(MediaTypeHeaderValues.ApplicationJson);
+            SupportedMediaTypes.Add(MediaTypeHeaderValues.TextJson);
+            SupportedMediaTypes.Add(MediaTypeHeaderValues.ApplicationAnyJsonSyntax);
+        }
+
+        /// <inheritdoc />
+        InputFormatterExceptionPolicy IInputFormatterExceptionPolicy.ExceptionPolicy => InputFormatterExceptionPolicy.MalformedInputExceptions;
+
+        /// <inheritdoc />
+        public override async Task<InputFormatterResult> ReadRequestBodyAsync(
+            InputFormatterContext context,
+            Encoding encoding)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (encoding == null)
+            {
+                throw new ArgumentNullException(nameof(encoding));
+            }
+
+            var httpContext = context.HttpContext;
+            var inputStream = GetInputStream(httpContext, encoding);
+
+            try
+            {
+                var result = await JsonSerializer.ReadAsync(inputStream, context.ModelType, _serializerOptions, httpContext.RequestAborted);
```

Comment:
I have a doubt about using `httpContext.RequestAborted` here. I'm not sure we use cancellation tokens in other places we do input reading because we generally want the request to go through if the client discconnected after sending the data.

Reply:
Yeah I think having the ct as a parameter to ReadRequestBodyAsync would be better. We don't do this in the FormReader.